sdk/communication/communication-phone-numbers/src/phoneNumbersClient.ts

```diff
@@ -110,9 +109,16 @@ export class PhoneNumbersClient {
       },
     };
 
+    this.client = new PhoneNumbersGeneratedClient(url, {
+      endpoint: url,
+      ...internalPipelineOptions,
+    });
     const authPolicy = createCommunicationAuthPolicy(credential);
-    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
-    this.client = new PhoneNumbersGeneratedClient(url, pipeline).phoneNumbers;
+    this.client.pipeline.addPolicy(authPolicy);
+
+    // These policies are temporary workarounds to address compatibility issues with Azure Core V2.
+    this.client.pipeline.addPolicy(phoneNumbersLroPolicy);
+    this.client.pipeline.addPolicy(createPhoneNumbersPagingPolicy(url));
```

Comment:
Since the `authPolicy` is being stored in a variable before being passed to the `addPolicy` method, I'd recommend you to keep the same pattern and have the `createPhoneNumbersPagingPolicy` result stored in a variable as well, even though this is a temporary workaround.